app/scripts/lib/tx-verification/tx-verification-middleware.ts

```diff
@@ -0,0 +1,111 @@
+import { hashMessage } from '@ethersproject/hash';
+import { verifyMessage } from '@ethersproject/wallet';
+import type { NetworkController } from '@metamask/network-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import { Json, JsonRpcParams, hasProperty, isObject } from '@metamask/utils';
+import {
+  JsonRpcRequest,
+  JsonRpcResponse,
+  JsonRpcEngineEndCallback,
+  JsonRpcEngineNextCallback,
+} from 'json-rpc-engine';
+import {
+  SIG_LEN,
+  TRUSTED_BRIDGE_SIGNER,
+} from '../../../../shared/constants/bridge';
+import { FIRST_PARTY_CONTRACT_NAMES } from '../../../../shared/constants/first-party-contracts';
+
+export type BridgeTxParams = {
+  chainId?: `0x${string}`;
+  data: string;
+  from: string;
+  to: string;
+  value: string;
+};
+
+/**
+ * Creates a middleware function that verifies bridge transactions from the
+ * Portfolio.
+ *
+ * @param networkController - The network controller instance.
+ * @returns The middleware function.
+ */
+export function createTxVerificationMiddleware(
+  networkController: NetworkController,
+) {
+  return function txVerificationMiddleware(
+    req: JsonRpcRequest<JsonRpcParams>,
+    _res: JsonRpcResponse<Json>,
+    next: JsonRpcEngineNextCallback,
+    end: JsonRpcEngineEndCallback,
+  ) {
+    if (
+      req.method !== 'eth_sendTransaction' ||
```

Comment:
We check this method in a few places such as the PPOM support, I wonder if it warrants a generic `ProviderMethod` enum or similar constant.

Reply:
I saw this as well, but decided to hold off on doing that within this PR.
I'd like to place such an enum within `eth-methods.ts`, however there would seem to be some overlap between `@metamask/keyring-api` and the `app.ts:MESSAGE_TYPE` and this new enum. In any event I opted for the `app.ts:MESSAGE_TYPE` and added two new entries.